identity/aziot-identityd/src/identity.rs

```diff
@@ -532,6 +541,7 @@ impl IdentityManager {
                         dps_auth_kind,
                         registration_id,
                         credentials,
+                        provisioning.local_gateway_hostname,
```

Comment:
@huguesBouvier - Does the proxy route HTTPS traffic to DPS endpoints correctly? I assume it will, however the X509 case will not work. Can you confirm?

https://github.com/Azure/iot-identity-service/blob/main/identity/aziot-identityd/src/identity.rs#L497-L507
If X509 doesn't work in nested mode, we need to failfast in `DpsAttestationMode::X509` case, possibly before grabbing the identity cert, by throwing an error.

Reply:
I believe the same issue will hit the IoT Hub `ManualAuthMethod::X509` case as well

Reply:
Right now we don't support DPS